wallet/local/wallet.go

```diff
@@ -134,101 +150,107 @@ func extractSubnetIDFromTx(tx interface{}) (ids.ID, error) {
+			newWallet, err := getWalletFromAccount(ctx, acc, w.defaultNetwork, &newConfig)
+			if err != nil {
+				return nil, fmt.Errorf("error refreshing wallet with new subnet ID: %w", err)
+			}
+			w.wallet = newWallet
+		}
 	}
-	w.Wallet = wallet
-	return nil
-}
 
-func (w *LocalWallet) Accounts() []account.Account {
-	return w.accounts
+	return w.wallet, nil
 }
 
-// CreateAccount creates a new Account using local key generation
-func (w *LocalWallet) CreateAccount() (*account.Account, error) {
-	newAccount, err := account.NewLocalAccount()
-	if err != nil {
-		return nil, fmt.Errorf("failed to create new Account: %w", err)
+// hasSeenSubnetID checks if a subnet ID has been seen before
+func (w *LocalWallet) hasSeenSubnetID(subnetID ids.ID) bool {
+	for _, id := range w.seenSubnetIDs {
+		if id == subnetID {
+			return true
+		}
 	}
-
-	// Add the Account to the wallet
-	w.AddAccount(newAccount)
-
-	return &newAccount, nil
+	return false
 }
 
-// ListAccounts returns all accounts managed by this wallet
-func (w *LocalWallet) ListAccounts() ([]*account.Account, error) {
-	// Return all accounts in the wallet
-	accounts := w.GetAllAccounts()
-	result := make([]*account.Account, len(accounts))
-	for i := range accounts {
-		result[i] = &accounts[i]
+// BuildTx constructs a transaction for the specified operation
+func (w *LocalWallet) BuildTx(ctx context.Context, params types.BuildTxParams) (types.BuildTxResult, error) {
+	// Validate parameters first
+	if err := params.Validate(); err != nil {
+		return types.BuildTxResult{}, fmt.Errorf("invalid parameters: %w", err)
 	}
-	return result, nil
-}
 
-// ImportAccount imports an existing Account into the wallet
-func (w *LocalWallet) ImportAccount(keyPath string) (*account.Account, error) {
-	// TODO: Implement Account import logic
-	// This would add the provided Account to the wallet
-	existingAccount, err := account.Import(keyPath)
+	walletToUse, err := w.getWalletToUse(ctx, params.AccountNames, &params)
 	if err != nil {
-		return nil, fmt.Errorf("error when importing Account %w", err)
+		return types.BuildTxResult{}, err
 	}
-	w.AddAccount(existingAccount)
-	return &existingAccount, nil
-}
 
-// BuildTx constructs a transaction for the specified operation
-func (w *LocalWallet) BuildTx(ctx context.Context, params types.BuildTxParams) (types.BuildTxResult, error) {
-	if err := w.loadAccountIntoWallet(ctx, params.Account, params.Network, &params); err != nil {
-		return types.BuildTxResult{}, fmt.Errorf("error loading account into wallet: %w", err)
+	var accountToUse account.Account
+	if len(params.AccountNames) > 0 {
+		accountToUse = w.accounts[params.AccountNames[0]]
+	} else {
+		accountToUse = w.accounts[w.activeAccountName]
 	}
-	return wallet.BuildTx(w.Wallet, params)
+
+	return wallet.BuildTx(walletToUse, accountToUse, params)
 }
 
 // SignTx signs a transaction
 func (w *LocalWallet) SignTx(ctx context.Context, params types.SignTxParams) (types.SignTxResult, error) {
```

Comment:
Sign, Build, Send requries no change at all

Reply:
functionality wise

Reply:
These changes are required for:

1. **ADR 5 implementation** - Support `AccountNames` parameter to temporarily use a different account for a single operation without changing the active account
2. **Refactored architecture** - Changed from passing `Account` in params to using `AccountNames` with active account fallback
3. **Efficiency optimization** - Subnet ID caching via `getWalletToUse()` to avoid unnecessary wallet rebuilds

The functionality is the same for simple cases, but these changes enable the new account management patterns and improve performance for multi-subnet operations.

Reply:
I dont agree, I think even with SignTx now supporting the concept of "active account":
func (w *LocalWallet) SignTx(ctx context.Context, params types.SignTxParams) (types.SignTxResult, error) {
General Steps are :
- get the account you want to use from params (whether active or not)
- loadAccountIntoWallet
- Sign